### ISSUE TITLE
Refresh Azure Discovery documentation

### ISSUE THREAD

**Issue by hugoShaka** [gravitational/teleport#31602]
## Applies To

https://goteleport.com/docs/server-access/guides/azure-discovery/

## Details

The documentation doesn't allow to setup a working discovery service:

- required permissions of the discovery service are not up-to-date
  ```json
  {
      "id": "/subscriptions/060a97ea-3a57-4218-9be5-dba3f19ff2b5/providers/Microsoft.Authorization/roleDefinitions/1f0c6014-1b0c-419a-854d-755df7399dcb",
      "properties": {
          "roleName": "TeleportDiscoveryHugo",
          "description": "Allows Teleport to discover Azure virtual machines",
          "assignableScopes": [
              "/subscriptions/060a97ea-3a57-4218-9be5-dba3f19ff2b5",
              "/subscriptions/060a97ea-3a57-4218-9be5-dba3f19ff2b5/resourceGroups/hugo-test-plan"
          ],
          "permissions": [
              {
                  "actions": [
                      "Microsoft.Compute/virtualMachines/read",
                      "Microsoft.Compute/virtualMachines/runCommand/action",
                      "Microsoft.Compute/virtualMachines/runCommands/write",
                      "Microsoft.Compute/virtualMachines/runCommands/read",
                      "Microsoft.Compute/virtualMachines/runCommands/delete"
                  ],
                  "notActions": [],
                  "dataActions": [],
                  "notDataActions": []
              }
          ]
      }
  }
  ```
- there's no mention of how to handle the case where the discovery VM has both system-assigned identities and user-assigned identities
- it doesn't clearly explain that region is required (else the discovery service runs silently without doing anything)
- It doesn't explain the prerequisites on the discovered VMs:
  - discovered VMs must have a single identity
  - their identity must grant the permissions to read all VMs (for azure join to work)
- (optional) it doesn't explain how to deal with multiple identities on the discovered machines and avoid sending a token to Teleport that inherits all the discovered machine permissions
- there's no explanation for troubleshooting (check if `run-command` is executed, the teleport status, ...)

## How will we know this is resolved?

Someone with no prior Azure joining nor Azure discovery knowledge can follow the guide and have a working setup.

**Comment by atburke** [comment]
Requiring the region is actually a bug, fixed by #32001.

**Comment by atburke** [comment]
Related: #28839
